src/Mvc/Mvc.Cors/src/CorsApplicationModelProvider.cs

```diff
@@ -30,8 +39,21 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            if (_mvcOptions.EnableEndpointRouting)
+            {
+                // When doing endpoint routing, translate IEnableCorsAttribute to an HttpMethodMetadata with CORS enabled.
+                ConfigureCorsEndpointMetadata(context.Result);
```

Comment:
Should this go in endpoint routing? Right now, you could configure `ICorsMetadata` for an endpoint, but that would not cause the `HttpMethodMetadata` to change to allow CORS

Reply:
I'm not sure what you mean *go in endpoint routing*. what would that look like?

Reply:
One of

1) Have `EndpointMetadata` complain if you have cors metadata but do not have `HttpMethodData.AllowCors` set when constructing the `Endpoint`
2) Have the builder update the `HttpMethodData` when it sees cors attributes
3) Remove `HttpMethodData.AllowCors` and use the presence of cors metadata to do CORS. Particularly feasible if `ICorsMetadata` is in abstractions.

The problem I have with this behavior now is that it's not enough to say here's CORS stuff for you endpoint, you have to explicitly update another metadata for it to work.

Reply:
I'll log and issue for this.

Reply:
https://github.com/aspnet/AspNetCore/issues/9127